File: t/Ch04_NavigationController/Ch03_TableVewController/EditorViewController.m
//
//  EditorViewController.m
//  Ch04_NavigationController
//
//  Created by ucom Apple Instructor on 2016/11/29.
//  Copyright © 2016年 ucom Apple Instructor. All rights reserved.
//

#import "EditorViewController.h"
//Delegation Step 1
@interface EditorViewController ()<UITextFieldDelegate>
@property (strong, nonatomic) IBOutlet UITextField *taskInput;
- (IBAction)saveHandler:(id)sender;

@end

@implementation EditorViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    //Delegation Step 2
    self.taskInput.delegate = self;
    if (self.row == -1){
        //New
        self.title = @"新增";
    }else{
        //Update
        self.title = @"編輯";
        NSArray * list = [NSArray arrayWithContentsOfFile:self.target];
        self.taskInput.text = list[self.row];
    }
}
//Delegation Step 3
-(BOOL)textFieldShouldReturn:(UITextField *)textField{
    [textField resignFirstResponder];
    return YES;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

- (IBAction)saveHandler:(id)sender {
    //收集使用者輸入
    NSString * task = self.taskInput.text;
    //更新Array
    NSMutableArray * list = [NSMutableArray arrayWithContentsOfFile:self.target];
    if (self.row == -1){
        [list addObject:task];
    }else{
        list[self.row] = task;
    }
    //更新檔案
    [list writeToFile:self.target atomically:YES];
    //回上一頁
    [self.navigationController popViewControllerAnimated:YES];
}
@end

File: ch08_DAO/ch08/EditorViewController.swift
//
//  EditorViewController.swift
//  ch08
//
//  Created by ucom Apple 13 on 2016/11/30.
//  Copyright © 2016年 Gillian_studio. All rights reserved.
//

import UIKit

class EditorViewController: UIViewController ,UITextFieldDelegate{

    var sid = 0
    var current = Contact()
    //textField
    @IBOutlet weak var photoImg: UIImageView!
    @IBOutlet weak var addressInput: UITextField!
    @IBOutlet weak var nameInput: UITextField!
    
    @IBAction func savehandler(sender: UIBarButtonItem) {
        current.name = nameInput.text!
        current.address = addressInput.text!
        
        current.photo =  UIImagePNGRepresentation( photoImg.image!) // UIImage -> NSData
        if sid == 0 {
            ContactDAO.insert(current)
        }else{
            ContactDAO.update(current)
        }
        self.navigationController?.popViewControllerAnimated(true)
    }
    
    
    @IBAction func phtotPickerahndler(sender: UISegmentedControl) {
    
    
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if sid == 0{
            self.title  = "New"
        }else
        {
            self.title = "Edit"
            current = ContactDAO.getContactBySid(sid)
            nameInput.text = current.name as String
            addressInput.text = current.address as String
            if current.photo != nil{
                photoImg.image = UIImage(data: current.photo!)
            }
        }
        
        nameInput.delegate  = self
        addressInput.delegate = self
        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        textField .resignFirstResponder()
        return true
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
